addons/account/models/res_partner_bank.py

```diff
@@ -32,6 +58,59 @@ def _check_journal_id(self):
             if len(bank.journal_id) > 1:
                 raise ValidationError(_('A bank account can belong to only one journal.'))
 
+    @api.constrains('allow_out_payment')
+    def _check_allow_out_payment(self):
+        """ Block enabling the setting, but it can be set to false without the group. (For example, at creation) """
+        for bank in self:
+            if bank.allow_out_payment:
+                if not self.user_has_groups('account.group_validate_bank_account'):
+                    raise ValidationError(_('You do not have the right to trust or un-trust a bank account.'))
+
+    @api.depends('partner_id', 'sanitized_acc_number', 'allow_out_payment')
+    def _compute_display_account_warning(self):
+        for bank in self:
+            if bank.allow_out_payment or not bank.sanitized_acc_number or bank.acc_type != 'iban':
+                bank.iban_warning = False
+                bank.money_transfer_warning = False
+                continue
+            bank_country = bank.sanitized_acc_number[:2]
+            bank.has_iban_warning = bank.partner_id.country_id and bank_country != bank.partner_id.country_id.code
+
+            bank_institution_code = bank.sanitized_acc_number[4:7]
+            money_transfer_service_map = bank._get_money_transfer_services()
+            bank.has_money_transfer_warning = money_transfer_service_map.get(bank_institution_code, False)
+
+    @api.depends('partner_id', 'sanitized_acc_number', 'allow_out_payment')
+    def _compute_display_account_warning_messages(self):
+        for bank in self:
+            bank.iban_warning = markupsafe.Markup(_("""
+                <span class="text-warning">Medium risk</span>: Iban <span class="fw-bold">{}</span>
+                is not from the same country as the partner ({}).<br/>
+                <a href="https://www.odoo.com/documentation/master/applications/finance/accounting/payables/pay/trusted_accounts.html">Check why it's risky.</a>
+            """)).format(bank.sanitized_acc_number, bank.partner_id.country_id.name)
+
+            bank_institution_code = bank.sanitized_acc_number[4:7]
+            money_transfer_service_map = bank._get_money_transfer_services()
+            money_transfer_service = money_transfer_service_map.get(bank_institution_code, False)
+            bank.money_transfer_warning = markupsafe.Markup(_("""
+                <span class="text-danger">High risk</span>: {} is a money transfer service and not a bank.
+                Double check if the account can be trusted by calling the vendor.<br/>
+                <a href="https://www.odoo.com/documentation/master/applications/finance/accounting/payables/pay/trusted_accounts.html">Check why.</a>
+            """)).format(money_transfer_service)
```

Comment:
These HTML snippets can actually be included in the view directly, the placeholders being the computed fields.